### ISSUE TITLE
Enforce the requirement to call gtk::init

### ISSUE THREAD

**Issue by gkoz** [gtk-rs/gtk#188]
It's unsafe to let the user do anything before calling `gtk::init`. See also #3.

I don't think we're going to make sweeping changes like in #34 just yet so for the time being I'm thinking of adding thread local booleans to `gdk` and `gtk`
```rust
thread_local!(static INITIALIZED: Cell<bool> = Cell::new(false));
```
and asserts to all constructors and free functions
```rust
assert!(INITIALIZED.with(|v| v.get()), "Not the main thread or failed to call `gtk::init`");
```

(Perhaps with another static to distinguish the two cases)

Unresolved questions: What about `idle_add`? Should this reach into `glib` as well?

**Comment by GuillaumeGomez** [comment]
It's hard to say. If we start to move functions from one lib to another, it can get confusing for users. Otherwise I agree with your other changes.

**Comment by gkoz** [comment]
Fixed in #189
